Core/Source/Themes/ThemeProvider.swift

```diff
@@ -3,8 +3,6 @@ import RxSwift
 import Logger
 
 public class ThemeProvider: ThemeProviderProtocol {
-    public typealias ThemeType = Theme
```

Comment:
I didn't actually know this would compile. I thought you had to explicitly define the `typealias` (in which case you'd have a naming conflict if the `ThemeProviderProtocol`'s `associatedtype` and the concrete theme type was `Theme`). It looks like the compiler is able to infer it from the `theme` property's type. Neat!

As long as you don't think it's confusing that there are two kinds named `Theme` in the project (the `associatedtype` on `ThemeProviderProtocol` in `Themer`, and the concrete `Theme` enum in `Core`), I think this solution is more elegant.

Reply:
I'm likely doing this in other places (explicitly defining the `typealias` when it could be inferred, and therefore being forced to name the `associatedtype` with `...Type` since it would otherwise cause a conflict).

Reply:
After some thought I do think it's confusing that these describe two different things (one is a an `associatedtype` while the other is a separate class), but they share the same name. Perhaps we should keep "Type" in the name, but remove the explicit definition of the `typealias` and let the compiler infer it?

An example where this causes issues is with `ViewModelType`. If you go through the project and attempt to rename `ViewModelType` to `ViewModel` you can see why this could lead to issues and get confusing.